motioneye/mjpgclient.py

```diff
@@ -123,39 +123,47 @@ def _check_error(self):
 
     def _error(self, error):
         logging.error('mjpg client for camera %(camera_id)s on port %(port)s error: %(msg)s' % {
-                'port': self._port, 'camera_id': self._camera_id, 'msg': utils.make_str(error)})
+            'port': self._port, 'camera_id': self._camera_id, 'msg': utils.make_str(error)})
 
         try:
             self.close()
 
         except Exception:
             pass
 
-    def _on_connect(self):
+    def _on_connect(self, future):
+        exc = future.exception()
+        if exc is not None and not isinstance(exc, IOStream.StreamClosedError):
+            future.result()
+
         logging.debug('mjpg client for camera %(camera_id)s connected on port %(port)s' % {
-                'port': self._port, 'camera_id': self._camera_id})
+            'port': self._port, 'camera_id': self._camera_id})
 
         if self._auth_mode == 'basic':
             logging.debug('mjpg client using basic authentication')
 
             auth_header = utils.build_basic_header(self._username, self._password)
-            self.write(b'GET / HTTP/1.1\r\nAuthorization: %s\r\nConnection: close\r\n\r\n' % auth_header)
+            self.write(b'GET /%(camera_id)/stream HTTP/1.1\r\nAuthorization: %(header)\r\nConnection: close\r\n\r\n' %
+                       {'camera_id': self._camera_id, 'header': auth_header})
 
         elif self._auth_mode == 'digest':  # in digest auth mode, the header is built upon receiving 401
-            self.write(b'GET / HTTP/1.1\r\n\r\n')
+            self.write(b'GET /%d/stream HTTP/1.1\r\n\r\n' % self._camera_id)
```

Comment:
@Mictronics 
These GET requests are the only change which is not applied the same or similar way in `dev` already. Since generally MJPG cameras and the MJPG stream from cam to browser work fine, can you explain the rationale behind this change?